### ISSUE TITLE
loaders.gl/las OOM Error  

### ISSUE THREAD

**Issue by gchoqueux** [visgl/loaders.gl#1912]
We use `loaders.gl/las` to load a large cloud with a multitude of laz files ([entwine cloud](https://entwine.io/)).
Then we visualize them simultaneously from hundreds of laz files with a library `THREE.js`.

After uploading hundreds of files, the `loaders.gl/las` throwns **OOM Error**.
It seems that this error is thrown when `emscripten` can't resize memory heap.

Is there a memory leak?
Is there a memory limit?
is possible to free the  `loaders.gl/las` memory?

**Comment by ibgreen** [comment]
If you are using the default setup, you are probably using the worker loader. 

You can set a flag to destroy the worker loader after each load: https://loaders.gl/docs/developer-guide/using-worker-loaders#disabling-reuse-of-workers. Perhaps using a fresh loader for each load will help?

**Comment by gchoqueux** [comment]
Thank you for the answer.

I don't the `load` method, I use directly the parser with the method `LASLoader.parse`.
The options `worker` and `reuseWorkers` are used in the parser?

**Comment by ibgreen** [comment]
No you would need to use `load()` or `parse()`.

**Comment by gchoqueux** [comment]
The `parse()` method fixes the `OOM error`.
But they are replaced by warnings from **las-worker.js** :

* _On Chrome :_ `Invalid asm.js: Unexpected token`
* _On Firefox :_ `asm.js type error: expecting argument type declaration for 'n' of the form 'arg = arg|0' or 'arg = +arg' or 'arg = fround(arg)'`  **las-worker.js:1:62997**

**Comment by ibgreen** [comment]
Haven't seen these. The LASLoader appears to work on the loaders.gl website:  https://loaders.gl/examples/pointcloud (and I can see the las worker has started in the Thread inspector). 

Are these just warnings? Do the point clouds actually load?

Not sure how to help...

<img width="616" alt="Screen Shot 2021-11-04 at 6 51 14 AM" src="https://user-images.githubusercontent.com/7025232/140325720-0aaf0926-9c5e-4708-8787-a3de32c1e62a.png">

**Comment by gchoqueux** [comment]
Yes, the point cloud files are loaded with `parse(data, LASLoader, { worker: true, reuseWorkers: false })` and worker has started.

As I load thousands of files, I get thousands of `asm/worker` warnings.

**Comment by gchoqueux** [comment]
ok, thanks

**Comment by greenwood-vaarst** [comment]
@gchoqueux did you ever resolve this? I'm having the same issue
